### ISSUE TITLE
Black square before sign "registered trademark"  (\textregistered)

### ISSUE THREAD

**Issue by mdk73** [plastex/plastex#39]
plasTeX renders a black square before the sign "registered trademark"  (\textregistered).

My test code:

``` LaTeX
\documentclass[a4paper,10pt,fleqn, twoside]{book}
\setlength{\textwidth}{15cm}
\setlength{\textheight}{23.5cm}
\setlength{\evensidemargin}{0cm}

%according Plastex reference section 4.1
\newif\ifplastex
\plastexfalse

\usepackage[bookmarks=true]{hyperref}

\ifplastex
%do nothing
\else
\hypersetup{
    unicode=true,          % non-Latin characters in Acrobat’s bookmarks
    pdffitwindow=false,     % window fit to page when opened
    pdfstartview={FitV},    % fits the height of the page to the window
    colorlinks=true,       % false: boxed links; true: colored links
    linkcolor=blue,          % color of internal links
    citecolor=green,        % color of links to bibliography
    filecolor=blue,      % color of file links
    urlcolor=blue           % color of external links
}
\fi

\begin{document}

%\chapter{Get the current Year}
Trademark\raisebox{1ex}{\tiny{\textregistered}}.

\end{document}
```

The shell output:

```
C:\Users\krause\workspace\manualUSB_en\plastex_test\trademark>c:\python27\python.exe c:\python27\Scripts\plastex --index
-columns=1 "--title=test case" .\plastex_trademarks.tex
plasTeX version 1.0
 ( c:\python27\lib\site-packages\plasTeX\Packages\book.pyc )....
 ( c:\python27\lib\site-packages\plasTeX\Packages\hyperref.pyc )......
INFO: Directing output files to directory: plastex_trademarks.
INFO: Importing templates from c:\python27\lib\site-
   packages\plasTeX\Renderers\PageTemplate
INFO: Importing templates from c:\python27\lib\site-
   packages\plasTeX\Renderers\XHTML
INFO: Importing templates from c:\python27\lib\site-
   packages\plasTeX\Renderers\XHTML\Themes\default
INFO: Using the imager "dvipng".
 [ index.html ]
WARNING: PIL (Python Imaging Library) is not installed.  Images will not be
   cropped.
Traceback (most recent call last):
  File "c:\python27\lib\site-packages\plasTeX\Imagers\__init__.py", line 737, in convert
    dest.crop()
  File "c:\python27\lib\site-packages\plasTeX\Imagers\__init__.py", line 265, in crop
    im, self.depth = self._stripBaseline(PILImage.open(self.path),
AttributeError: 'NoneType' object has no attribute 'open'
WARNING: failed to crop C:\Users\krause\workspace\manualUSB_en\plastex_test
   \trademark\plastex_trademarks\images/img-0001.png ('NoneType' object has no
   attribute 'open')
```

Just in case the warning related to PIL has something todo with it: I have the fork Pillow installed, as PIL is not anymore actively developed (at least to my knowledge). Theoretically it should work the same...

The PDF-output from LaTeX and the html-output from plasTeX are attached.

[plastex_trademarks.pdf](https://github.com/tiarno/plastex/files/401459/plastex_trademarks.pdf)
[plastex_trademarks.zip](https://github.com/tiarno/plastex/files/401460/plastex_trademarks.zip)

**Comment by PatrickMassot** [comment]
You need to install [pil-compat](https://pypi.python.org/pypi/pil-compat/1.0.0).

You can also directly use the unicode ®, plasTeX won't complain and the output will be much nicer.

**Comment by mdk73** [comment]
pil-compat worked, thanks.

Using ® directly I did not get to work, this approach was not successful:

```LaTex
Trademark
\ifplastex
\raisebox{1ex}{®}
\else
\raisebox{1ex}{\tiny{\textregistered}}
\fi
```

And this neither:
```LaTex
Trademark
\ifplastex
®
\else
\raisebox{1ex}{\tiny{\textregistered}}
\fi
```

**Comment by PatrickMassot** [comment]
The second one works here. What does the output html file look like?

The first one (with ``raisebox``) cannot work. The ``raisebox`` command is purely about form or rendering. It's bound to depend strongly on the renderer. As far as I can see, the default (XHTML) renderer of plasTeX will ignore it unless its content has been rendered to an image. And indeed it's hard to see what kind of html it should generate.

On the other hand you could expect ``\textsuperscript`` to work. However the default renderer has no template for it. You can patch ``plasTeX/Renderers/XHTML/Sentences.zpts`` by adding the following lines:
```xml
name: textsuperscript
<sup tal:content="self">superscripted text</sup>
```
I will probably create a pull request for this if we don't hear anything from @kesmit13 or @tiarno on this thread.

Alternatively you can put the last line in a file ``textsuperscript.zpt`` in your working directory and tell plasTeX where to find it by setting the ``XHTMLTEMPLATES`` environment variable.

Note that this won't solve the problem if the second attempt does not work.

**Comment by mdk73** [comment]
The attached zip-file containes a latex-testfile, the pdf-output and the html-output generated on my system.
```\raisebox{1ex}{\tiny{\textregistered}}``` is working, though not looking very pretty.

I did not patch my system, because it would lead to code that works (on patched systems) or not (on not patched systems). Personally I consider this too hard to maintain...

[trademark.zip](https://github.com/tiarno/plastex/files/479761/trademark.zip)

**Comment by PatrickMassot** [comment]
This was fixed in #40.
